app/viewModels/documents/AddAnotherDocumentViewModel.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewModels.documents
+
+import config.FrontendAppConfig
+import models.DocType.{Previous, Support, Transport}
+import models.{ArrivalId, ConsignmentLevelDocuments, DocType, Index, Mode, RichOptionalJsArray, UserAnswers}
+import pages.documents.{DocumentReferenceNumberPage, TypePage}
+import pages.sections.documents.DocumentsSection
+import play.api.i18n.Messages
+import play.api.mvc.Call
+import viewModels.{AddAnotherViewModel, ListItem}
+
+case class AddAnotherDocumentViewModel(
+  listItems: Seq[ListItem],
+  onSubmitCall: Call,
+  nextIndex: Index,
+  docTypeList: Seq[DocType],
+  allowMore: Boolean
+) extends AddAnotherViewModel {
+
+  override val prefix: String = "document.addAnotherDocument"
+
+  override def maxCount(implicit config: FrontendAppConfig): Int = config.maxSupportingDocuments + config.maxSupportingDocuments
+
+  override def maxLimitLabel(implicit messages: Messages): String = messages(s"$prefix.maxLimit.label")
+
+  def maxLimitLabelForType(implicit config: FrontendAppConfig, messages: Messages): String = {
+    val groupedByType = docTypeList.groupBy(identity)
+    if (groupedByType.getOrElse(Support, Seq.empty).length == config.maxSupportingDocuments) {
+      messages(s"$prefix.maxLimitForType.label", Support.display.toLowerCase, Transport.display.toLowerCase)
+    } else {
+      messages(s"$prefix.maxLimitForType.label", Transport.display.toLowerCase, Support.display.toLowerCase)
+    }
+  }
+
+  def limitReachedForType(implicit config: FrontendAppConfig): Boolean = {
+    val groupedByType = docTypeList.groupBy(identity)
+    groupedByType.getOrElse(Support, Seq.empty).length == config.maxSupportingDocuments ||
+    groupedByType.getOrElse(Transport, Seq.empty).length == config.maxTransportDocuments
+  }
```

Comment:
I wonder if perhaps there is no need for both of these? Perhaps `maxLimitLabelForType` could return an Option[String], then in the view do a `maxLimitLabelForType.getOrElse(maxLimitLabel)` or something similar